src/components/Card.js

```diff
@@ -1,58 +1,122 @@
         this._imageWindow = this._element.querySelector('.card__image');
         this._imageWindow.addEventListener('click', () => this._handleImageClick({link: this._link, name: this._name}));
     }
 
-    _handleLikeIcon = () => {
-        // Add active class to card's like button
-        this._likeButton.classList.toggle('card__like-button_active');
+    // Check if the card is liked by the user
+    isLiked() {
+        return this._cardLikes.some((cardLike) => {
+            return cardLike._id === this._userId;
+        })
+    }
+
+    // Update the card's number of likes
+    updateLikeCount(likes) {
+        this._cardLikes = likes;
+        return this._likeCounter.textContent = likes.length;
+    }
+
+    // Display the number of likes
+    displayLikeCount() {
+        return this._likeCounter.textContent = this._cardLikes.length;
+    }
+
+    // Display the like icon based on its status
+    displayLikeIcon() {
+        if (this.isLiked()) {
+            this._likeButton.classList.add('card__like-button_active');
+        } else {
+            this._likeButton.classList.remove('card__like-button_active');
+        }
+    }
+
+    // Activate the like icon
+    addLikeIcon() {
+        this._likeButton.classList.add('card__like-button_active');
+    }
+
+    // Deactivate the like icon
+    removeLikeIcon() {
+        this._likeButton.classList.remove('card__like-button_active');
     }
 
-    _handleDeleteCard() {
+    // Display bin icon on cards created by the user
+    _handleDeleteButton() {
+
+        // Check if the user id matches the card owner's id
+        if (this._userId !== this._cardOwnerId) {
+
+            // If it does, remove the delete button
+            this._deleteButton.remove();
+        }
+    }
+
+    // Handle the deletion of cards
+    handleDeleteCard() {
         // Remove element from the DOM on click
         this._element.remove();
 
         // Remove the link to the DOM element
         this._element = null;
     }
 
+    // Get card template
     _getTemplate() {
-
-        // Get card template
+        
         return document
             .querySelector(this._cardSelector)
             .content.querySelector('.card')
             .cloneNode(true);
     }
 
+    // Render the card
     getView() {
         
         // Create card template
         this._element = this._getTemplate();
+
+        
+        // Display the bin icon on the user's cards
+        this._deleteButton = this._element.querySelector('.card__delete-button');
+        this._handleDeleteButton();
+        this._deleteButton.addEventListener('click', () => this._handleDeleteClick());
         
+        // Display the number of likes
+        this._likeCounter = this._element.querySelector('.card__like-number');
+        this.displayLikeCount();
+
+        // Change the like button color
+        this._likeButton = this._element.querySelector('.card__like-button');
+        this.displayLikeIcon();
```

Comment:
The code for rendering likes gets duplicated. The code should be reused when you initializing the card and when updating likes. Card should have one method for updating likes and one method for rendering likes.

The code for rendering likes should look like this:

```js
// in Card class 

// call this method from index.js after successful api response in `then` block
setLikes(likes) { 
   // set instance variable
    this._likes = likes; 
    this._renderLikes(); 
} 

isLiked() {
 // return true if user liked the card, otherwise false
}

_renderLikes() {
  //set  likes counter content using this._likes.length

  if (this.isLiked()) {
    // add active class to the like button
  } else {
    // remove active class from the like button
  }
}

createCard() { 
  ...
  // render initial likes, you saved them in the constructor
  this._renderLikes()
}

// index.js
if (card.isLiked()) {
  // delete like
} else {
 // add like
}
```

Reply:
Thank you for the detailed explanation! I've cleaned up my code accordingly.